### PR TITLE
[CORE-1477] - As a user I have a basic CLI.

### PR DESCRIPTION
**Pull request by forgems**
Implemented features for now:
- login with email and password and some kind of token store
- push / pull classes
- push / pull scripts
- design configuration file (syncano.yml)

### CODE REVIEW

File: syncano_sync/classes.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
+# coding=UTF8
+from __future__ import print_function, unicode_literals
+
+import re
+
+from . import LOG, mute_log
+from syncano.models.classes import Class
+
+ALLOWED_TYPES = set(("array", "boolean", "datetime", "file", "float",
```
Comment (23doors) [inline syncano_sync/classes.py]: what about relation, geopoint?

File: syncano_sync/scripts.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+import re
+from collections import defaultdict
+
+from syncano.models import fields
+from syncano.models.incentives import Script
+
+from . import LOG, mute_log
+
+
+ALLOWED_RUNTIMES = {
```
Comment (23doors) [inline syncano_sync/scripts.py:13]: why validate it like that on CLI level?

Comment (forgems) [inline syncano_sync/scripts.py:13]: It's mostly used for getting file extension, I will remove the check. But there is also a problem with syncano-python that validates to much on client side (eg runtime_name validator is outdated).

File: syncano_sync/__init__.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import logging
+from contextlib import contextmanager
+
+
+LOG = logging.getLogger('syncano-sync')
+SYNCANO_LOG = logging.getLogger('syncano')
+
+
+@contextmanager
+def mute_log(logger=None):
```
Comment (23doors) [inline syncano_sync/__init__.py:10]: I guess syncano library log should probably be a little less verbose by default. It's especially visible in tests of syncano lib which are full of messy logs.

File: syncano_sync/classes.py (hunk 1 above)
Comment (23doors) [inline syncano_sync/classes.py]: I guess relation is not added yet on python library part?

File: syncano_sync/classes.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                     "integer", "object", "reference", "string", "text",
+                     "geopoint"))
+
+ALLOWED_PERMISIONS = ('none', 'read', 'create_objects')
+
+# FIXME: We don't require it on backend so, we should not require it in library
+Class._meta.get_field('schema').required = False
```
Comment (23doors) [inline syncano_sync/classes.py:17]: Be sure to create tickets for things that need to be fixed in the library

File: syncano_sync/__init__.py (hunk 3 above)
Comment (forgems) [inline syncano_sync/__init__.py:10]: It issues warnings when saving objects with read_only fields set, fetched from syncano. I reported this issue for @opalczynski :)

File: syncano_sync/classes.py (hunk 1 above)
Comment (23doors) [inline syncano_sync/classes.py]: Also, why not declaring a set like {'array', 'boolean' ... }?

**Comment by 23doors** [comment]
left a few comments but overall, looks like a good start! 
You don't like line breaks in code, do you? ;) A lot of "wall of text" sections ;) 
Anyway, we should add some basic circleci tests possibly with flake8/isort checks, but this might as well go into a separate PR.

**Comment by 23doors** [comment]
👍

**Pull request merged**